pkg/collector/corechecks/snmp/report/report_device_metadata.go

```diff
@@ -43,67 +39,121 @@ func (ms *MetricSender) ReportNetworkDeviceMetadata(config *checkconfig.CheckCon
 	}
 }
 
-func buildNetworkDeviceMetadata(deviceID string, idTags []string, config *checkconfig.CheckConfig, store *valuestore.ResultValueStore, tags []string, deviceStatus metadata.DeviceStatus) metadata.DeviceMetadata {
-	var vendor, sysName, sysDescr, sysObjectID string
+func buildMetadata(metadataConfigs checkconfig.MetadataConfig, values *valuestore.ResultValueStore) *metadata.Store {
```

Comment:
nit: we could rename to `buildMetadataStore` to avoid being too generic, especially as we also have `buildNetworkDeviceMetadata`

Reply:
Sure, renamed here 1bf7094c8